framework/src/kernels/ADTimeDerivative.C

```diff
@@ -15,18 +15,21 @@ InputParameters
 ADTimeDerivative::validParams()
 {
   InputParameters params = ADTimeKernelValue::validParams();
-  params.addClassDescription("The time derivative operator with the weak form of $(\\psi_i, "
-                             "\\frac{\\partial u_h}{\\partial t})$.");
+  params.addClassDescription(
+      "The time derivative operator with the weak form of $(\\psi_i, "
+      "\\alpha \\dot \\frac{\\partial u_h}{\\partial t})$, where \\alpha is the coefficient.");
+  params.addRequiredParam<MaterialPropertyName>("coefficient",
+                                                "The coefficient of the first time derivative");
   return params;
 }
 
 ADTimeDerivative::ADTimeDerivative(const InputParameters & parameters)
-  : ADTimeKernelValue(parameters)
+  : ADTimeKernelValue(parameters), _coef(getADMaterialProperty<Real>("density"))
 {
 }
 
 ADReal
 ADTimeDerivative::precomputeQpResidual()
 {
-  return _u_dot[_qp];
+  return _coef[_qp] * _u_dot[_qp];
```

Comment:
I'd prefer it if you derived a class instead of adding a required new parameter to a fundamental framework class (which is going to break input files all over the place). We have a kernel like this called SusceptibilityTimeDerivative in phase field.